src/CargoBuild/CargoBuildTask.cs

```diff
@@ -0,0 +1,437 @@
+                    if (!exited)
+                    {
+                        process.Kill();
+                        Log.LogError($"Killed process after max timeout reached : '{info.FileName}'");
+                        return -1;
+                    }
+
+                    return process.ExitCode;
+                });
+
+                processTask.Start();
+                return (ExitCode)await processTask;
+            }
+            catch (Exception ex)
+            {
+                Log.LogWarningFromException(ex);
+                return await Tasks.Task.FromResult(ExitCode.Failed);
+            }
+        }
+
+        private async Task<bool> DownloadRustupAsync()
+        {
+            if (File.Exists(_rustUpInitBinary) && await VerifyInitHashAsync())
+            {
+                return true;
+            }
+            else if (File.Exists(_rustUpInitBinary))
+            {
+                // If the hash doesn't match, that likely means there is a new version of rustup-init.exe available.
+                File.Delete(_rustUpInitBinary);
+            }
+
+            string rustupDownloadLink = _rustupDownloadLink;
+            Log.LogMessage(MessageImportance.Normal, $"Downloading -- {rustupDownloadLink}");
+            using (var client = new HttpClient())
+            {
+                Task<HttpResponseMessage> response = client.GetAsync(rustupDownloadLink);
+                if (!Directory.Exists(_rustInstallPath))
+                {
+                    Directory.CreateDirectory(_rustInstallPath);
+                }
+
+                using var responseStream = new FileStream(_rustUpInitBinary, FileMode.CreateNew);
+                HttpResponseMessage res = await response;
+                await res.Content.CopyToAsync(responseStream);
+            }
+
+            return await VerifyInitHashAsync();
+        }
+
+        private async Task<bool> InstallRust()
+        {
+            var rustupBinary = UseMsRustUp ? _msRustupBinary : _rustUpBinary;
+            if (File.Exists(_cargoPath) && Directory.Exists(_rustupHome) && File.Exists(_rustUpBinary) || File.Exists(_cargoHome) && UseMsRustUp != true || File.Exists(_msRustupBinary))
+            {
+                return false;
+            }
+
+            if (UseMsRustUp)
+            {
+                string? workingDirPart = new DirectoryInfo(BuildEngine.ProjectFileOfTaskNode).Parent?.Parent?.FullName;
+                if (Directory.Exists(workingDirPart))
+                {
+                    Log.LogMessage(MessageImportance.Normal, "Installing MS Rustup");
+                    string sdkRootPath = Path.Combine(workingDirPart!, "content\\dist");
+                    ExecuteProcessAsync("powershell.exe", $".\\msrustup.ps1 '{_cargoHomeBin}'", sdkRootPath, _envVars).GetAwaiter().GetResult();
+                }
+            }
+
+            foreach (var envVar in _envVars)
+            {
+                Environment.SetEnvironmentVariable(envVar.Key, envVar.Value);
```

Comment:
This changes the env vars for the MSBuild.exe process. This should be avoided.

Instead, change the environment of the child process you're launching.